pkg/build/api/v1/types.go

```diff
@@ -371,6 +375,13 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// RunBash holds a bash script to be executed in a container running the
+	// build output image.
+	RunBash string `json:"runBash,omitempty" description:"bash script to be executed in a container running the build output image"`
```

Comment:
> allows easy editing

We might want to have a field in the Web Console for this, and we would not be able to add that experience if it is just a path to a file in an external source repo.

I intentionally did not tag this PR for UI team nor API review because I wanted some feedback from @bparees first. It is really WIP for now.


> and in most cases improves trackability since we don't have an audit trail?

That's a good point. @smarterclayton @bparees what do you think about this? Is it a problem, do we need trackability of what is the command to run tests?